### PR TITLE
Move dummy_operator.py to dummy.py (#11178)

### PR DESCRIPTION
**Pull request by kishvanchee**
This change is related to #11178 for the `dummy_operator`

**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by kishvanchee** [comment]
@turbaszek @mik-laj Please review the changes.

**Comment by kishvanchee** [comment]
I'm not sure how to fix **CI Build / Backport packages (pull_request)**. The error states 

```
Traceback (most recent call last):
  File "/import_all_provider_classes.py", line 60, in import_all_provider_classes
    _module = importlib.import_module(module_name)
  File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/lib/python3.6/site-packages/airflow/providers/qubole/example_dags/example_qubole.py", line 24, in <module>
    from airflow.operators.dummy import DummyOperator
ModuleNotFoundError: No module named 'airflow.operators.dummy'
```
but this should have failed for imports from other places too right? I don't see this error popping up in the breeze environment.

```
root@b20475852b92:/opt/airflow# python
Python 3.6.12 (default, Sep 10 2020, 18:03:29) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from airflow.operators.dummy import DummyOperator as d1
>>> from airflow.operators.dummy_operator import DummyOperator as d2
>>> d2.__name__
'DummyOperator'
>>> d1.__name__
'DummyOperator'
```

**Comment by mik-laj** [comment]
@kishvanchee c The files in the providers directory must be either compatible with Airflow 1.10.13, or they must be automatically refactor to be compatible. When we release Airflow 1.10.13 this problem will disappear. Details: https://github.com/apache/airflow/issues/11178
https://github.com/apache/airflow/issues/10152 Migration to Airflow 2.0
Now you can either revert all changes to the providers directory or add auto-refactor (preferred). To do it, add a new tuple here: https://github.com/apache/airflow/blob/9514cfe64b35289566c4e56c491c36b43c4fb737/backport_packages/refactor_backport_packages.py#L145-L155

**Comment by kishvanchee** [comment]
@mik-laj Got it thanks. I'll make that change. How does this affect only certain operators and not all operators? Since so far I see only bash and python are incorporated, and now we'll incorporate dummy too.

**Comment by kishvanchee** [comment]
@mik-laj please review the changes

**Review by turbaszek — CHANGES_REQUESTED** [review]
Please update occurrences of import of DummyOperator in docs so they use new path not deprecated one

**Comment by kishvanchee** [comment]
@turbaszek Can you please confirm? I was excluding `docs/_api` from my search since I thought these would get generated automatically. I guess from your comment I'm wrong.

**Comment by turbaszek** [comment]
We should fix all occurrences 😉

**Comment by turbaszek** [comment]
@kishvanchee can you rebase please?

**Comment by turbaszek** [comment]
@kishvanchee can you please rebase due to conflicts?

**Review by turbaszek — APPROVED** [review]
LGTM 🆗

**Comment by kaxil** [comment]
Can you please rebase your PR on latest Master since we have applied [Black](https://github.com/apache/airflow/commit/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68) and [PyUpgrade](https://github.com/apache/airflow/commit/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5) on Master.

It will help if your squash your commits into single commit first so that there are less conflicts.

**Comment by turbaszek** [comment]
@kishvanchee any update on this one?

**Comment by kishvanchee** [comment]
@turbaszek @kaxil Hi, I apologize for the delay. I'll update this by tomorrow afternoon IST.

**Comment by kishvanchee** [comment]
@kaxil @turbaszek Can you please have a look now? I have updated, rebased and squashed the commits.

**Review by turbaszek — APPROVED** [review]
LGTM! The failed test is related to scheduler tests

**Review by ashb — CHANGES_REQUESTED** [review]
This needs a note added to UPDATING.md and UPGRADING_TO_2.0.md please

**Comment by kishvanchee** [comment]
> This needs a note added to UPDATING.md and UPGRADING_TO_2.0.md please

Can you please point out where exactly? In UPDATING.md I found a section ** Changes to import paths **. I couldn't find something similar in UPGRADING_TO_2.0.md.

**Comment by ashb** [comment]
Oh yes, there's nothing in there (perhaps there should be) -- so just UPDATING.md will be fine

**Comment by kishvanchee** [comment]
@ashb can you please have a look?

File: UPDATING.md [hunk 1]
```diff
@@ -219,8 +219,15 @@ with third party services to the ``airflow.providers`` package.
...
 
+According to [AIP-21](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-21%3A+Changes+in+import+paths)
+`_operator` suffix has been removed from operators. A deprecation warning has also been raised for paths
+importing with the suffix.
 
+The following table shows changes in import paths.
 
+| Old path                            | New path                   |
+|-------------------------------------|----------------------------|
+| airflow.operators.dummy_operator.py | airflow.operators.dummy.py |
```
Comment (ashb) [inline UPDATING.md]: I'm surprised to see we don't have more info here/a table like this already!

Comment (ashb) [reply UPDATING.md]: There's this block further on.

```
#### Removed deprecated import mechanism

The deprecated import mechanism has been removed so the import of modules becomes more consistent and explicit.

For example: `from airflow.operators import BashOperator`
becomes `from airflow.operators.bash_operator import BashOperator`

```

Comment (ashb) [inline UPDATING.md]:
```suggestion
| airflow.operators.dummy_operator.DummyOperator | airflow.operators.dummy.DummyOperator |
```

**Review by ashb — APPROVED** [review]

**Comment by ashb** [comment]
@kishvanchee Could you rebase this please? And check if there are any new instances that have "snuck in" in the mean time.

**Comment by kishvanchee** [comment]
@ashb Yes sure, I will do this tomorrow. Apologies for the delay.

**Comment by ashb** [comment]
> @ashb Yes sure, I will do this tomorrow. Apologies for the delay.

The delay was all us, no need to apologise!

**Comment by kaxil** [comment]
I think this needs updating https://github.com/apache/airflow/blob/23e743ec09bffe30a9a2b17cc873af5ef9d3cd9f/dev/provider_packages/refactor_provider_packages.py#L141-L145 too

**Pull request merged**